### PR TITLE
Support non-planar CVPixelBuffer

### LINKED ISSUE #2: Doesn't encode nonplanar buffer
I'm trying to encode `depthMap` retrieved from [ARDepthData](https://developer.apple.com/documentation/arkit/ardepthdata) but it doesn't seem to work.

### PR DESCRIPTION
**Pull request by longseespace**
Fix #2

### CODE REVIEW

**Comment by s1ddok** [comment]
thanks for this! I'll try to get it merged soon

File: Sources/SwiftyCVPixelBuffer/CVPixelBuffer+Utils.swift [hunk 1]
```diff
@@ -75,26 +77,73 @@ public extension CVPixelBuffer {
...
+                for _ in 0..<height {
+                    memcpy(startOfRowDest, startOfRowSrc, min(bytesPerRowSrc, bytesPerRowDest))
+                    startOfRowSrc = startOfRowSrc?.advanced(by: bytesPerRowSrc)
+                    startOfRowDest = startOfRowDest?.advanced(by: bytesPerRowDest)
+                }
+            }
+
+        } else {
+            for plane in 0 ..< pixelBufferPlaneCount {
+                let dest        = CVPixelBufferGetBaseAddressOfPlane(copy, plane)
```
Comment (eugenebokhan) [inline Sources/SwiftyCVPixelBuffer/CVPixelBuffer+Utils.swift]: strange alignment

File: Sources/SwiftyCVPixelBuffer/CVPixelBuffer+Utils.swift [hunk 2]
```diff
@@ -75,26 +77,73 @@ public extension CVPixelBuffer {
...
-            let source = self.baseAddress(of: plane)
-            let height = self.height(of: plane)
-            let bytesPerRow = self.bytesPerRow(of: plane)
-            memcpy(dest, source, height * bytesPerRow)
+
+        let pixelBufferPlaneCount: Int = CVPixelBufferGetPlaneCount(self)
+
+
+        if pixelBufferPlaneCount == 0 {
+            let dest = CVPixelBufferGetBaseAddress(copy)
```
Comment (eugenebokhan) [inline Sources/SwiftyCVPixelBuffer/CVPixelBuffer+Utils.swift]: can we use `copy.baseAdress`?

File: Sources/SwiftyCVPixelBuffer/CVPixelBuffer+Utils.swift [hunk 3]
```diff
@@ -75,26 +77,73 @@ public extension CVPixelBuffer {
...
+
+        let width = CVPixelBufferGetWidth(self)
+        let height = CVPixelBufferGetHeight(self)
+        let formatType = CVPixelBufferGetPixelFormatType(self)
+        let attachments = CVBufferGetAttachments(self, .shouldPropagate)
+
+        CVPixelBufferCreate(nil, width, height, formatType, attachments, &_copy)
+
+        guard let copy = _copy else {
+            throw Error.allocationFailed
```
Comment (eugenebokhan) [inline Sources/SwiftyCVPixelBuffer/CVPixelBuffer+Utils.swift]:
```suggestion
     else { throw Error.allocationFailed }
```

File: Sources/SwiftyCVPixelBuffer/CVPixelBuffer+Utils.swift [hunk 4]
```diff
@@ -75,26 +77,73 @@ public extension CVPixelBuffer {
...
+                let height      = CVPixelBufferGetHeightOfPlane(self, plane)
+                let bytesPerRowSrc = CVPixelBufferGetBytesPerRowOfPlane(self, plane)
+                let bytesPerRowDest = CVPixelBufferGetBytesPerRowOfPlane(copy, plane)
+
+                if bytesPerRowSrc == bytesPerRowDest {
+                    memcpy(dest, source, height * bytesPerRowSrc)
+                } else {
+                    var startOfRowSrc = source
+                    var startOfRowDest = dest
+                    for _ in 0..<height {
```
Comment (eugenebokhan) [inline Sources/SwiftyCVPixelBuffer/CVPixelBuffer+Utils.swift]:
```suggestion
                    for _ in 0 ..< height {
```

Comment (eugenebokhan) [reply Sources/SwiftyCVPixelBuffer/CVPixelBuffer+Utils.swift]: it's inconsistent with [this](https://github.com/s1ddok/SwiftyCVPixelBuffer/blob/4f2f0deb8cd5274692010e6c2f9a183e0d4f9e69/Sources/SwiftyCVPixelBuffer/CVPixelBuffer%2BUtils.swift#L90)

File: Sources/SwiftyCVPixelBuffer/CVPixelBuffer+Codable.swift [hunk 5]
```diff
@@ -108,26 +116,36 @@ extension CVPixelBuffer: CustomEncodable, CustomDecodable {
...
-
+        
         pixelBuffer.lockBaseAddress(options: [])
         defer { pixelBuffer.unlockBaseAddress(options: []) }
 
-        for plane in 0 ..< planeCount {
-            let dest = pixelBuffer.baseAddress(of: plane)
-            let source = try planesData.decode(Data.self,
-                                               forKey: .planeDataKey(index: plane))
+        if (isPlanar) {
```
Comment (s1ddok) [inline Sources/SwiftyCVPixelBuffer/CVPixelBuffer+Codable.swift]: let's remove brackets here

File: Sources/SwiftyCVPixelBuffer/CVPixelBuffer+Utils.swift [hunk 6]
```diff
@@ -120,4 +169,11 @@ extension CVPixelBuffer {
         return .init(bytes: source,
                      count: height * bytesPerRow)
     }
+    
+    func data() -> Data? {
+        guard let source = self.baseAddress else { return nil }
+        let height = self.height
+        let bytesPerRow = CVPixelBufferGetBytesPerRow(self)
```
Comment (s1ddok) [inline Sources/SwiftyCVPixelBuffer/CVPixelBuffer+Utils.swift]: let's use `self.bytesPerRow`

File: Sources/SwiftyCVPixelBuffer/CVPixelBuffer+Utils.swift (hunk 1 above)
Comment (s1ddok) [reply Sources/SwiftyCVPixelBuffer/CVPixelBuffer+Utils.swift]: also, let's use extensions instead of C-style funcs

**Pull request merged**